Tests/SwiftLintFrameworkTests/BaselineTests.swift

```diff
@@ -0,0 +1,225 @@
+            insertionIndex: 2
+        )
+    }
+
+    func testViolationsWithNoFile() throws {
+        try testViolationDetection(
+            violations: Self.violations.map { $0.with(location: Location(file: nil)) },
+            lineShift: 0,
+            newViolationRuleDescription: ArrayInitRule.description,
+            insertionIndex: 2
+        )
+    }
+
+    func testViolationDetection() throws {
+        let violations = [
+            ArrayInitRule.description,
+            BlockBasedKVORule.description,
+            ArrayInitRule.description,
+            ClosingBraceRule.description,
+            ClosingBraceRule.description,
+            ClosingBraceRule.description,
+            BlockBasedKVORule.description,
+            DirectReturnRule.description,
+            ArrayInitRule.description,
+            ClosingBraceRule.description
+        ].violations
+
+        let ruleDescriptions = [
+            ArrayInitRule.description,
+            BlockBasedKVORule.description,
+            ClosingBraceRule.description,
+            DirectReturnRule.description
+        ]
+
+        for ruleDescription in ruleDescriptions {
+            for insertionIndex in 0..<violations.count {
+                try testViolationDetection(
+                    violations: violations,
+                    newViolationRuleDescription: ruleDescription,
+                    insertionIndex: insertionIndex
+                )
+            }
+            break
+        }
+    }
+
+    func testCompare() throws {
+        try testBlock {
+            let oldViolations = Array(Self.violations.dropFirst())
+            let newViolations = Array(Self.violations.dropLast())
+            let oldBaseline = Baseline(violations: oldViolations)
+            let newBaseline = Baseline(violations: newViolations)
+            XCTAssertEqual(oldBaseline.compare(newBaseline), [Self.violations.first])
+            XCTAssertEqual(newBaseline.compare(oldBaseline), [Self.violations.last])
+        }
+    }
+
+    private func testViolationDetection(
+        violations: [StyleViolation],
+        lineShift: Int = 1,
+        newViolationRuleDescription: RuleDescription,
+        insertionIndex: Int
+    ) throws {
+        try testBlock {
+            let baseline = Baseline(violations: violations)
+            var newViolations = lineShift != 0
+                ? try violations.lineShifted(by: lineShift, path: sourceFilePath)
+                : violations
+            let line = ((insertionIndex + 1) * 2) - 1 + lineShift
+            let violation = StyleViolation(
+                ruleDescription: newViolationRuleDescription,
+                location: Location(file: sourceFilePath, line: line, character: 1)
+            )
+            newViolations.insert(violation, at: insertionIndex)
+            XCTAssertEqual(baseline.filter(newViolations), [violation])
+        }
+    }
+
+    private func testBlock(_ block: () throws -> Void) throws {
```

Comment:
Ideally, the path to the example path is something local. So I propose to manage file creation and cleanup only in here and pass the path to the caller: 
```suggestion
    private func withExampleFileCreated(_ block: (filePath: String) throws -> Void) throws {
```